Manage multichain addresses (#8351)
rotki/rotki: docs/api.rst

   :statuscode 409: An account given to edit does not exist or a given tag does not exist.
   :statuscode 500: Internal rotki error

> Chain agnostic account operations
> ==================================
> 
> .. http:patch:: /api/(version)/blockchains/agnostic/accounts

Reviewer: Name. and format.

```suggestion
Account operations by chain type
==================================

.. http:patch:: /api/(version)/blockchains/(chaintype)/accounts
```

So make the chaintype be part of the URL string. Should only `evm`. `bitcoin` or `substrate`

   .. note::

>       Supported blockchains types: ``EVM, BITCOIN, SUBSTRATE, EVM_LIKE``

Reviewer: From the api's point of view we need EVM and EVM like to be one choice. Touching lefteris.eth in all evm and evmlike chains.

   .. note::
      This endpoint doesn't support ENS resolution

>    Doing a PATCH on the blockchains endpoint with a list of accounts to edit will edit the label and tags for those accounts in all the blockchains where they are tracked.

Reviewer: ```suggestion
   Doing a PATCH on this endpoint with a list of accounts to edit will edit the label and tags for those accounts in all the chains of the same type where they are tracked.
```

 
   **Example Request**:

   .. http:example:: curl wget httpie python-requests

>       PATCH /api/1/blockchains/evm/accounts HTTP/1.1

Reviewer: ```suggestion
      PATCH /api/1/blockchains/type/evm/accounts HTTP/1.1
```

>       PATCH /api/1/blockchains/agnostic/accounts HTTP/1.1

Reviewer: Same changes in format and text here and below

      Host: localhost:5042
      Content-Type: application/json;charset=UTF-8

      {
          "accounts": [
              {
                  "address": "0x78b0AD50E768D2376C6BA7de33F426ecE4e03e0B",
                  "label": "my new metamask",
                  "tags": [
                      "public",
                      "metamask"
                  ]
              },
              {
                  "address": "0xc37b40ABdB939635068d3c5f13E7faF686F03B65",
                  "label": "my hardware wallet"
              }
          ]
      }

>    :reqjson list[object] accounts: A list of account data to edit.

Reviewer: Shouldn't you mention the blockchain type here too?

Author: We don't do it for other endpoints since it is a variable in the URL and I've documented in the note the valid types

   :reqjsonarr string address: The address of the account to edit. It doesn't allow ENS names.
   :reqjsonarr string[optional] label: An optional label to edit for the account. Cannot be empty string.
   :reqjsonarr list[optional] tags: An optional list of tags to attach to the account. Can be null. Should never be an empty list.

   **Example Response**:

   .. sourcecode:: http

      HTTP/1.1 200 OK
      Content-Type: application/json

      {
          "result" : True,
          "message": "",
      }

>    :resjson list result: Okey result if all the accounts were edited correctly.

Reviewer: ```suggestion
   :resjson list result: Okay result if all the accounts were edited correctly.
```

   :statuscode 200: Accounts successfully edited
   :statuscode 400: Provided JSON or data is in some way malformed. Given list to edit is empty.
   :statuscode 401: User is not logged in.
   :statuscode 409: An account given to edit does not exist or a given tag does not exist.
   :statuscode 500: Internal rotki error

> .. http:delete:: /api/(version)/blockchains/(chain_type)/accounts

Reviewer: Do it like above. We need consistency here

> .. http:delete:: /api/(version)/blockchains/agnostic/accounts

Reviewer: same here

   .. note::
      Supported blockchains types: ``EVM, BITCOIN, SUBSTRATE``

   .. note::
      This endpoint doesn't support ENS resolution

   Doing a DELETE on this endpoint with a list of accounts in the json data will remove these accounts from the tracked accounts for the all the chains of the ``chain_type`` provided.
   If one of the given accounts to add is invalid the entire request will fail.

   **Example Request**:

   .. http:example:: curl wget httpie python-requests

>       DELETE /api/1/blockchains/evm/accounts HTTP/1.1

Reviewer: change this too
